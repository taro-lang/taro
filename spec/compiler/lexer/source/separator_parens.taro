def value() : Unit = hello() 
var value = here()
a() + b()
a()() + b()()
((()))