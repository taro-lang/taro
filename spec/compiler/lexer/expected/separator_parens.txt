Def (def) line: 0, length: 3, position: (0,2)
Space ( ) line: 0, length: 1, position: (3,3)
Identifier (value) line: 0, length: 5, position: (4,8)
LParen (() line: 0, length: 1, position: (9,9)
RParen ()) line: 0, length: 1, position: (10,10)
Space ( ) line: 0, length: 1, position: (11,11)
Colon (:) line: 0, length: 1, position: (12,12)
Space ( ) line: 0, length: 1, position: (13,13)
Identifier (Unit) line: 0, length: 4, position: (14,17)
Space ( ) line: 0, length: 1, position: (18,18)
Assign (=) line: 0, length: 1, position: (19,19)
Space ( ) line: 0, length: 1, position: (20,20)
Identifier (hello) line: 0, length: 5, position: (21,25)
LParen (() line: 0, length: 1, position: (26,26)
RParen ()) line: 0, length: 1, position: (27,27)
Space ( ) line: 0, length: 1, position: (28,28)
Var (var) line: 1, length: 3, position: (0,2)
Space ( ) line: 1, length: 1, position: (3,3)
Identifier (value) line: 1, length: 5, position: (4,8)
Space ( ) line: 1, length: 1, position: (9,9)
Assign (=) line: 1, length: 1, position: (10,10)
Space ( ) line: 1, length: 1, position: (11,11)
Identifier (here) line: 1, length: 4, position: (12,15)
LParen (() line: 1, length: 1, position: (16,16)
RParen ()) line: 1, length: 1, position: (17,17)
Identifier (a) line: 2, length: 1, position: (0,0)
LParen (() line: 2, length: 1, position: (1,1)
RParen ()) line: 2, length: 1, position: (2,2)
Space ( ) line: 2, length: 1, position: (3,3)
Plus (+) line: 2, length: 1, position: (4,4)
Space ( ) line: 2, length: 1, position: (5,5)
Identifier (b) line: 2, length: 1, position: (6,6)
LParen (() line: 2, length: 1, position: (7,7)
RParen ()) line: 2, length: 1, position: (8,8)
Identifier (a) line: 3, length: 1, position: (0,0)
LParen (() line: 3, length: 1, position: (1,1)
RParen ()) line: 3, length: 1, position: (2,2)
LParen (() line: 3, length: 1, position: (3,3)
RParen ()) line: 3, length: 1, position: (4,4)
Space ( ) line: 3, length: 1, position: (5,5)
Plus (+) line: 3, length: 1, position: (6,6)
Space ( ) line: 3, length: 1, position: (7,7)
Identifier (b) line: 3, length: 1, position: (8,8)
LParen (() line: 3, length: 1, position: (9,9)
RParen ()) line: 3, length: 1, position: (10,10)
LParen (() line: 3, length: 1, position: (11,11)
RParen ()) line: 3, length: 1, position: (12,12)
LParen (() line: 4, length: 1, position: (0,0)
LParen (() line: 4, length: 1, position: (1,1)
LParen (() line: 4, length: 1, position: (2,2)
RParen ()) line: 4, length: 1, position: (3,3)
RParen ()) line: 4, length: 1, position: (4,4)
RParen ()) line: 4, length: 1, position: (5,5)